misk-hibernate-testing/src/main/kotlin/misk/jdbc/TestDatabasePool.kt

```diff
@@ -66,7 +67,12 @@ class TestDatabasePool(
 
   /** Return a config-specific pool, creating it if necessary. */
   internal fun getPool(config: DataSourceConfig): ConfigSpecificPool {
-    val key = config.database!!
+    var key = config.database!!
+    for (pool in poolsByKey.values) {
+      if (key.startsWith(pool.key)) {
```

Comment:
This class is used in `DataSourceService` which modifies the config by the replacing the original database name with the one returned by the `takeDatabase()` method of this class.

So we start with a config like: `database: mydb` and then end up with a config like `database:mydb_20190806_1`. The pool key is the database name from the original config, so we have to match on that to prevent creating a new pool on each call.

Reply:
Hopefully that makes sense!